packages/core/src/workers/password-encryption-worker.ts

```diff
@@ -0,0 +1,36 @@
+import path from 'node:path';
+import { fileURLToPath } from 'node:url';
+
+import { findUp } from 'find-up';
+import { Tinypool } from 'tinypool';
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = path.dirname(__filename);
+
+// Use findUp to locate the build directory
+const buildDirectory = await findUp('build', { type: 'directory', cwd: __dirname });
+
+if (!buildDirectory) {
+  throw new Error('Build directory not found.');
+}
+
+/**
+ * User password encryption worker pool.
+ *
+ * @remarks
+ * This worker pool is used to encrypt user password with Argon2i encryption method.
+ * Since the encryption process is CPU intensive,
+ * to better protect the main thread I/O performance,
+ * we use separate thread threads to handle the encryption process.
+ */
+const passwordEncryptionWorker = new Tinypool({
+  filename: path.join(buildDirectory, 'workers/tasks/argon2i.js'),
+  maxThreads: 2,
+  // By default the worker will be terminated immediately after the task is completed.
+  // Since starting and terminating a worker thread can lead to some performance overhead,
+  // Set the idle timeout to 5 seconds will keep the worker thread alive for concurrent requests.
+  // See {@link https://piscinajs.dev/api-reference/Instance/#constructor-new-piscinaoptions} for more details
```

Comment:
update to tinypool url?

Reply:
Tinypool adopts and extends most of its APIs from PiscinaJS. For this specific configuration, Tinypool does not provide its own parameter definitions, so you’ll need to refer to the PiscinaJS API documentation for details.